### ISSUE TITLE
Text base direction again — but now with a solution:-)

### ISSUE THREAD

**Issue by iherman** [w3c/pub-manifest#75]
The issue of base direction has been plagued us for a while (see also [issue 354 on wpub](https://github.com/w3c/wpub/issues/354)), but we may have a proper solution for it now. There has been discussions [elsewhere](https://w3c.github.io/rdf-dir-literal/) to look for a solution, a (failed) attempt to revive the discussion in RDF land and, finally, a [breakout session at TPAC](https://www.w3.org/2019/09/18-rdf-dir-literal-minutes.html). This led the JSON-LD WG to re-open the [issue of adding base direction to JSON-LD 1.1](https://github.com/w3c/json-ld-syntax/issues/11). On its F2F meeting at TPAC, the JSON-LD WG has accepted a series of resolutions, see the minutes of [Thursday](https://www.w3.org/2018/json-ld-wg/Meetings/Minutes/2019/2019-09-19-json-ld) and [Friday](https://www.w3.org/2018/json-ld-wg/Meetings/Minutes/2019/2019-20-19-json-ld) for the details.

The essence is: JSON-LD 1.1 will introduce a new keyword `@direction` that can be used, essentially, the same way as `@language`: can be part of the context to denote a global (ie, default) value, and can also be used as part of an individual literal. In our terminology, a `LocalizableString` in our manifest can have a directional value, if needed, just the same way as we handle language tags. 

_What I propose is to make changes on the manifest specification to adopt this feature._ What it would mean for the manifest is:

- the current `direction` term disappears and is replaced by a `@direction` keyword usage alongside `@language` (see [2.5.1](https://w3c.github.io/pub-manifest/#manifest-lang-dir-global))
- [item specific settings (in 2.5.2)](https://w3c.github.io/pub-manifest/#manifest-lang-dir-local) will include `@direction`
- The [definition of `LocalizableString`](https://w3c.github.io/pub-manifest/#LocalizableString) will be updated
- Some additions to the context file to empower all this.

The result, I believe, would be a _much_ cleaner format for our manifest, and we can but this issue at rest at last.

There is one caveat, though. This is a _JSON-LD 1.1 feature_, i.e., we become dependent on JSON-LD 1.1. This, by itself, is not a problem, JSON-LD 1.1 is slightly ahead of use in its advance towards a Rec. However, we have to be careful to use this feature in a way that it does not upset JSON-LD 1.0 processors that are supposed to simply ignore an unknown keyword. Without going into details what this means that we should _not_ create the `direction` alias for `@direction`, as we do for, e.g., `@language` and `@value`. As a consequence, _I would propose that we remove the usage of these aliases, and we use `@langauge` and `@value`; otherwise it looks very inconsistent_.

As soon as the JSON-LD 1.1 editor's draft includes the new feature, I will put in a PR to adopt these changes in the manifest document. We can decide, through the PR, whether we agree with these changes.

(This should supersede, ie, close, issue #39, and also make the discussions in https://github.com/w3c/wpub/issues/354 moot. Finally, it should close the gap in the i18n review in #38.)

Cc: @r12a @aphillips @mattgarrish @wareid @BigBlueHat @laudrain @llemeurfr @GarthConboy @rdeltour @dauwhe

**Comment by iherman** [comment]
For the RDF inclined among you the question may be what RDF statements are generated for this feature. The [JSON-LD WG Meeting minutes' section](https://www.w3.org/2018/json-ld-wg/Meetings/Minutes/2019/2019-09-20-json-ld#section3) may provide the necessary answers for now (before the publication of the new ed draft).

**Comment by iherman** [comment]
It would be nice to have a real-life example in the document. Have anybody such an example that I could use? Ie, a title that has possible directional issues and that should have an explicit base direction assigned.

@BigBlueHat you referred to scientific articles in Hebrew or Arabic that mix their script with latin scientific terms...

@r12a any idea?

**Comment by r12a** [comment]
In our [string-meta](https://w3c.github.io/string-meta/#an-example) doc we use the following example, which i believe is a real book found by Addison. Strip the markup and you should have something useful.

```
"title": "<span lang='ar' dir='rtl'>HTML و CSS: تصميم و إنشاء مواقع الويب</span>",
```

When displayed, this should look like this (read it right-to-left) (i couldn't figure out how to right-align it). Note that the 4 directional runs are in reversed order in the example vs expected display.  
<span lang='ar' dir='rtl'>HTML و CSS: تصميم و إنشاء مواقع الويب</span>

To help with cut & paste and to see the sequence of characters, follow [this link to UniView](https://r12a.github.io/uniview/?charlist=HTML%20%D9%88%20CSS:%20%D8%AA%D8%B5%D9%85%D9%8A%D9%85%20%D9%88%20%D8%A5%D9%86%D8%B4%D8%A7%D8%A1%20%D9%85%D9%88%D8%A7%D9%82%D8%B9%20%D8%A7%D9%84%D9%88%D9%8A%D8%A8).

Does that work?

**Comment by iherman** [comment]
@r12a thanks! I have not yet started the PR, but that should work.

Just to use it as a common example with multilingual title, what would be the title in English (or French)?

Thanks!

**Comment by aphillips** [comment]
From the string-meta source:

```
<!--
 Title below is actually "HTML and CSS: Design and Build Websites"
 ASIN: 1118871642
 ISBN-13: 978-1118871645
 ISBN-10: 1118871642
-->
```

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/publishing/groups/publ-wg/Meetings/Minutes/2019/2019-10-07-pwg#section1).

- `RESOLVED:  ivan to merge direction PR`
<details><summary><i class="differentiate">View the transcript</i></summary>
Directionality<br/>
Wendy Reid: <a href="https://github.com/w3c/pub-manifest/issues/75">https://github.com/w3c/pub-manifest/issues/75</a><br/>
Wendy Reid: first issue is about directionality<br/>
Ivan Herman: See Pull request for directionality<br/>
Ivan Herman: I have already reported that this whole direction issue now has a clear solution<br/>
… I’ve made a PR<br/>
… you can do direction much like you can do language<br/>
… with the same syntax, either globally or for a given member<br/>
… so the PR fixes all the bits<br/>
… there is one issue which I put into the document<br/>
… having direction in the system means that we have a dependency on JSON LD 1.1<br/>
… which is running in parallel with our spec; they plan CR at the same time, they have implementations.<br/>
… so they may be ahead of us<br/>
… but if you want to run a manifest through a JSON-LD processor that only supports 1.0 you will get errors<br/>
… so I created a separate context file that doesn’t have direction<br/>
… and made an appendix in the doc<br/>
… so you can use this context if you don’t need direction, and need to use a JSON-LD 1.0 processor<br/>
… so the question to the group is that it’s OK to keep this as an appendix in the document<br/>
… or put it somewhere else, or not do it at all and depend on JSON-LD 1.1<br/>
… so I want help from the group in deciding what to do<br/>
Wendy Reid: any questions?<br/>
Brady Duga: I don’t have strong feelings, but it does feel strange to put in the spec itself<br/>
… most things that process this won’t be JSON-LD processors<br/>
… but I’m not against it<br/>
… I think it should be in a read me or impl guide<br/>
Matt Garrish: I’m concerned about how this might be interpreted<br/>
… changing our context which is required seems weird<br/>
… it could add confusion to have an alternate context<br/>
Ivan Herman: I put it there without being convinced it’s a good idea<br/>
… I’m fine removing it<br/>
… so where would this information go?<br/>
… we could give it to the testing people and have this added to information about testing<br/>
Matt Garrish: I would put in our wiki for implementation details, so it’s discoverable<br/>
Wendy Reid: a wiki is good<br/>
Dave Cramer: This seems like it would be a helpful note in the spec itself, if this language property is used, mention that JSON-LD 1.1 is required<br/>
… it’s a good use of non-normative text<br/>
Ivan Herman: any other comments?<br/>
… one comment… for those of you who have been looking at this<br/>
… if you look at this example<br/>
… it is a good example there for real cases where base direction is necessary<br/>
… some people were wondering if this stuff was necessary<br/>
… We found a book with a title in arabic and english<br/>
Ivan Herman: so it’s good to have a real-world example<br/>
Matt Garrish: I think it’s useful to have direction; we have it in EPUB.<br/>
Ivan Herman: is there a consensus to merge?<br/>
Matt Garrish: +1<br/>
Wendy Reid: +1<br/>
Brady Duga: +1<br/>
Dave Cramer: +1<br/>
Gregorio Pellegrino: +1<br/>
Ivan Herman: we should also pledge to never talk about direction again :)<br/>
Resolution #2: ivan to merge direction PR<br/>
</details>
